assessments/projects/openfga/self-assessment.md

```diff
@@ -0,0 +1,277 @@
+  
+**Writing an Authorization Model**
+
+  - Semantic Model Verification: OpenFGA validates that Authorization Models are semantically valid, avoiding cyclical or problematic definitions and other disallowed criteria.
+
+  - Version and Configuration: When writing a model, a new version is created (models are immutable). Applications must be configured to use the new version after validation and confirmation of expected behavior.
+
+**Calling the Authorization Query endpoints**
+
+When the [/check](https://openfga.dev/api/service#/Relationship%20Queries/Check) and [/list-objects](https://openfga.dev/api/service#/Relationship%20Queries/ListObjects) endpoints are called, OpenFGA limits the number of simultaneous paths explored and enforces depth limitations on the graph traversal. 
+
+To protect against DoS attacks, OpenFGA restricts both the number of simultaneous paths explored and the depth of paths traversed in the graph.
+
+**Upgrading OpenFGA Database Schema**
+
+Database Migration Planning: When installing new versions of OpenFGA, database migrations can be executed in a way that minimizes downtime and ensure a smooth transition in the target system.
+
+### Goals
+
+* Simplify and standardize authorization processes, making them more consistent across various applications and systems.
+
+* Establish patterns and standards for externalized authorization.
+
+* Create architectural patterns, terminologies, and protocols that enable interoperability among different authorization systems.
+
+* Deliver an authorization service for any application component.
+
+* Enable centralized authorization decisions and permits diverse teams to implement authorization using a shared framework across various application components.
+
+### Non-Goals
+
+* Tools for management of groups/roles/permissions not inherently provided to the end-users.
+* Does not intend to serve as a comprehensive data repository for non-authorization related data.
+* Does not aim to provide a complete authentication and Access Control Solution.
+
+## Self-Assessment Use
+
+This self-assessment is created by the OpenFGA team to perform an internal analysis of the project's security. It is not intended to provide a security audit of OpenFGA, or function as an independent assessment or attestation of OpenFGA's security health.
+
+This document serves to provide OpenFGA users with an initial understanding of OpenFGA's security, where to find existing security documentation, OpenFGA plans for security, and general overview of OpenFGA security practices, both for development of OpenFGA as well as security of OpenFGA.
+
+This document provides the CNCF TAG-Security with an initial understanding of OpenFGA to assist in a joint-assessment, necessary for projects under incubation. Taken together, this document and the joint-assessment serve as a cornerstone for if and when OpenFGA seeks graduation and is preparing for a security audit.
+
+## Security Functions and Features

```

Comment:
I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. This section needs more details in the security function and features available for the project, the granularity and details are needed. This talks about projects role as security provider, but this section needs to talk about how the project is achieving itself security goals through the functions and features available.